includes/database/engine/class-table.php

```diff
@@ -2,133 +2,146 @@
 /**
  * Base Custom Database Table Class.
  *
- * @package     EDD
- * @subpackage  Database\Tables
- * @copyright   Copyright (c) 2018, Easy Digital Downloads, LLC
- * @license     http://opensource.org/licenses/gpl-2.0.php GNU Public License
- * @since       3.0
+ * @package     Database
+ * @subpackage  Table
+ * @copyright   Copyright (c) 2020
+ * @license     https://opensource.org/licenses/gpl-2.0.php GNU Public License
+ * @since       1.0.0
  */
 namespace EDD\Database;
 
 // Exit if accessed directly
 defined( 'ABSPATH' ) || exit;
 
 /**
- * A base WordPress database table class, which facilitates the creation of
- * and schema changes to individual database tables.
+ * A base database table class, which facilitates the creation of (and schema
+ * changes to) individual database tables.
  *
  * This class is intended to be extended for each unique database table,
- * including global multisite tables and users tables.
+ * including global tables for multisite, and users tables.
  *
- * It exists to make managing database tables in WordPress as easy as possible.
+ * It exists to make managing database tables as easy as possible.
  *
  * Extending this class comes with several automatic benefits:
  * - Activation hook makes it great for plugins
  * - Tables store their versions in the database independently
  * - Tables upgrade via independent upgrade abstract methods
  * - Multisite friendly - site tables switch on "switch_blog" action
  *
- * @since 3.0
+ * @since 1.0.0
  */
 abstract class Table extends Base {
 
 	/**
-	 * @var string Table name, without the global table prefix
+	 * Table name, without the global table prefix.
+	 *
+	 * @since 1.0.0
+	 * @var   string
 	 */
 	protected $name = '';
 
 	/**
-	 * @var string Optional description.
+	 * Optional description.
+	 *
+	 * @since 1.0.0
+	 * @var   string
 	 */
 	protected $description = '';
 
 	/**
-	 * @var mixed Database version
+	 * Database version.
+	 *
+	 * @since 1.0.0
+	 * @var   mixed
 	 */
 	protected $version = '';
 
 	/**
-	 * @var boolean Is this table for a site, or global
+	 * Is this table for a site, or global.
+	 *
+	 * @since 1.0.0
+	 * @var   bool
 	 */
 	protected $global = false;
 
 	/**
-	 * @var string Passed directly into register_activation_hook()
+	 * Passed directly into register_activation_hook()
+	 *
+	 * @since 1.0.0
+	 * @var   string
 	 */
-	protected $file = EDD_PLUGIN_FILE;
+	protected $file = '';
```

Comment:
Do we not still need `EDD_PLUGIN_FILE` here?

Reply:
We do. I missed this in the merge.

Reply:
I did more digging into `$file` usage in Berlin, and it appears like this variable can be removed completely.

1. Activation hooks are an unreliable way to perform database upgrades
2. In this instance `EDD_PLUGIN_FILE` does not have a `maybe_upgrade()` method to call anyways

Reply:
Alright, fine by me.